scripts/build-installer-unix.sh

```diff
@@ -0,0 +1,146 @@
+#   0. 'git'
+#   1. 'curl'
+#   2. 'nix-shell'
+#   3. 'stack'
+
+set -e
+
+usage() {
+    test -z "$1" || { echo "ERROR: $*" >&2; echo >&2; }
+    cat >&2 <<EOF
+  Usage:
+    $0 DAEDALUS-VERSION CARDANO-BRANCH OS OPTIONS*
+
+  Build a Daedalus installer.
+
+  OS is either 'linux' or 'osx'.
+
+  Options:
+    --build-id BUILD-NO       Identifier of the build; defaults to '0'
+
+    --travis-pr PR-ID         Travis pull request id we're building
+    --nix-path NIX-PATH       NIX_PATH value
+
+    --upload-s3               Upload the installer to S3
+
+    --verbose                 Verbose operation
+    --quiet                   Disable verbose operation
+    
+EOF
+    test -z "$1" || exit 1
+}
+alias   argnz='test $# -ge 1 -a ! -z "$1" || usage "empty value for"'
+alias  arg2nz='test $# -ge 2 -a ! -z "$2" || usage "empty value for"'
+alias  argnzf='test $# -ge 1 -a ! -z "$1" || fail "missing value for"'
+alias arg2nzf='test $# -ge 2 -a ! -z "$2" || fail "missing value for"'
+fail() { echo "ERROR: $*" >&2; exit 1;
+}
+retry() {
+        local tries=$1; argnzf "iteration count"; shift
+        for i in $(seq 1 ${tries})
+        do if "$@"
+           then return 0
+           fi
+           sleep 5s
+        done
+        fail "persistent failure to exec:  $*"
+}
+
+###
+### Argument processing
+###
+verbose=true
+build_id=0
+travis_pr=true
+upload_s3=
+
+daedalus_version="$1"; argnz "product version"; shift
+cardano_branch="$1";   argnz "Cardano SL branch to build Daedalus with"; shift
+os="$1";               argnz "OS to build for"; shift
+
+case "${os}" in
+        linux ) key=linux.p12;;
+        osx )   key=macos.p12;;
+        * )     usage "Unsupported OS provided: ${os}";;
+esac
+
+set -u ## Undefined variable firewall enabled
+while test $# -ge 1
+do case "$1" in
+           --build-id )       arg2nz "build identifier";    build_id="$2"; shift;;
+           --travis-pr )      arg2nz "Travis pull request id";
+                                                           travis_pr="$2"; shift;;
+           --nix-path )       arg2nz "NIX_PATH value";
+                                                     export NIX_PATH="$2"; shift;;
+           --upload-s3 )                                   upload_s3=t;;
+
+           ###
+           --verbose )        echo "$0: --verbose passed, enabling verbose operation"
+                                                             verbose=t;;
+           --quiet )          echo "$0: --quiet passed, disabling verbose operation"
+                                                             verbose=;;
+           --help )           usage;;
+           "--"* )            usage "unknown option: '$1'";;
+           * )                break;; esac
+   shift; done
+
+set -e
+if test -n "${verbose}"
+then set -x
+fi
+
+mkdir -p ~/.local/bin
+export PATH=$HOME/.local/bin:$PATH
+
+if test "${os}" = "linux"
+then
+        echo "INFO: running 'sudo mount -o remount,exec,size=4G,mode=755 /run/user'"
```

Comment:
This part maybe should stay part of travis-ci since it's travis-ci specific.

It could be possible someone has `noexec` or very small `/run/user`, but otherwise it will confuse people.